flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveDatabaseUtil.java

```diff
@@ -44,7 +43,9 @@ static Database instantiateHiveDatabase(String databaseName, CatalogDatabase dat
 
         Map<String, String> properties = database.getProperties();
 
-        boolean isGeneric = isGenericForCreate(properties);
+        properties.putIfAbsent(CatalogPropertiesUtil.IS_GENERIC, "true");
```

Comment:
why set is_generic=true here?

Reply:
If the database properties don't have the `is_generic` key, it's considered generic by default.

Reply:
This might make sense during the process of `createDatabase` (This method is only called in HiveCatalog#createDatabase), but it doesn't make sense in this utility class. If anyone else used such util method in other places, I don't think they will expect such behavior.

Reply:
I have updated the PR and we don't need to treat flink and hive databases differently.